tests/unit/completion/test_models.py

```diff
@@ -939,3 +941,73 @@ def bench():
         model.set_pattern('ex 123')
 
     benchmark(bench)
+
+
+@pytest.fixture
+def tab_with_history(fake_web_tab, tabbed_browser_stubs, info, monkeypatch):
+    """Returns a fake tab with some fake history items."""
+    tab = fake_web_tab(QUrl('https://github.com'), 'GitHub', 0)
+    current_idx = 2
+    monkeypatch.setattr(
+        tab.history, 'current_idx',
+        lambda: current_idx,
+    )
+
+    history = []
+    for url, title in [
+            ("http://example.com/index", "list of things"),
+            ("http://example.com/thing1", "thing1 detail"),
+            ("http://example.com/thing2", "thing2 detail"),
+            ("http://example.com/thing3", "thing3 detail"),
+            ("http://example.com/thing4", "thing4 detail"),
+    ]:
+        entry = mock.Mock(spec=QWebEngineHistoryItem)
```

Comment:
Hmm, Qt [doesn't seem to like that](https://travis-ci.org/qutebrowser/qutebrowser/jobs/427092467).

Reply:
Apparently this works nowadays (even with Qt 5.7), no idea why, but hey, I'm happy that way... :slightly_smiling_face: